eZ/Publish/Core/Persistence/Legacy/Content/Location/Handler.php

```diff
@@ -310,20 +310,55 @@ public function copySubtree($sourceId, $destinationParentId)
             );
         }
 
-        // If subtree root is main location for its content, update subtree section to the one of the
-        // parent location content
-        $subtreeRootContentInfo = $this->contentHandler->loadContentInfo($copiedSubtreeRootLocation->contentId);
-        if ($subtreeRootContentInfo->mainLocationId === $copiedSubtreeRootLocation->id) {
-            $this->setSectionForSubtree(
-                $copiedSubtreeRootLocation->id,
-                $this->contentHandler->loadContentInfo($this->load($destinationParentId)->contentId)->sectionId
-            );
-        }
+        $destinationParentSectionId = $this->getSectionId($destinationParentId);
+        $this->updateSubtreeSectionIfNecessary($copiedSubtreeRootLocation, $destinationParentSectionId);
 
         return $copiedSubtreeRootLocation;
     }
 
     /**
+     * Retrieves section ID of the location's content.
+     *
+     * @param int $locationId
+     *
+     * @return int
+     */
+    private function getSectionId($locationId)
+    {
+        $location = $this->load($locationId);
+        $locationContentInfo = $this->contentHandler->loadContentInfo($location->contentId);
+
+        return $locationContentInfo->sectionId;
+    }
+
+    /**
+     * If the location is the main location for its content, updates subtree section.
+     *
+     * @param Location $location
+     * @param int $sectionId
+     */
+    private function updateSubtreeSectionIfNecessary(Location $location, $sectionId)
+    {
+        if ($this->isMainLocation($location)) {
+            $this->setSectionForSubtree($location->id, $sectionId);
```

Comment:
It currently doesn't check that. In the legacy, it actually did, but only for the source node, not all of its children. This makes me wonder if it wouldn't be better to not check the section, because of differences in how it would be handled. What I mean is, if we introduce the check:
* If the sections of the source node and the destination node don't match, then the new section is set for the whole subtree.
* If the sections of the source node and the destination node do match, then the subtree would be untouched, meaning it could be anything. This would, in my opinion, create inconsistency. It would be certainly easier to explain if moving the location always resulted in changing the section of the whole subtree to the new one.
@glye what's your opinion?